torch/quantization/ns/numeric_suite_core_apis_fx.py

```diff
@@ -0,0 +1,241 @@
+    stats: List[torch.Tensor]
+
+    def __init__(
+        self,
+        node_name: str,
+        model_name: str,
+        other_node_name: Optional[str] = None,
+    ):
+        super().__init__()
+        self.stats: List[torch.Tensor] = []
+        # name of the node whose output this Logger is capturing
+        self.node_name = node_name
+        # name of the model from which the node originated from
+        self.model_name = model_name
+        # name of the other node with a matching Logger
+        # used to link node_a_copy -> logger_a to node_c -> logger_c
+        # in a_shadows_b
+        self.other_node_name = other_node_name
+
+    def forward(self, x: torch.Tensor):
+        self.stats.append(x.detach())
+        return x
+
+    def __repr__(self):
+        return f"OutputLogger(node_name={self.node_name}, model_name={self.model_name}, other_node_name={self.other_node_name})"
+
+# Note: this is not a user facing API
+# TODO(future PR): wrap this in a user facing API which does not
+#   expose FX types.
+def prepare_model_outputs(
+    name_a: str,
+    gm_a: GraphModule,
+    name_b: str,
+    gm_b: GraphModule,
+    logger_cls: Callable,
+) -> Tuple[GraphModule, GraphModule]:
+
+    matched_node_pairs = get_matching_node_pairs(gm_a, gm_b)
+
+    nodes_to_instrument_a = []
+    nodes_to_instrument_b = []
+    for match_name, (node_a, node_b,) in matched_node_pairs.items():
+        # TODO(future PR): do not observe pairs of nodes we do not care
+        #   about (both fp32, denylist, etc)
+        nodes_to_instrument_a.append(node_a)
+        nodes_to_instrument_b.append(node_b)
+
+    gm_a = remove_observers_add_loggers(gm_a, nodes_to_instrument_a, logger_cls, name_a)
+    gm_b = remove_observers_add_loggers(gm_b, nodes_to_instrument_b, logger_cls, name_b)
+    return (gm_a, gm_b)
+
+# Note: this is not a user facing API
+# TODO(future PR): wrap this in a user facing API which does not
+#   expose FX types.
+# TODO(future PR): align on naming
+# this is equivalent of just the comparison extraction part of `ns.compare_model_outputs`
+def get_matching_activations(
+    gm_a: GraphModule,
+    gm_b: GraphModule,
+    logger_cls: Callable,
+) -> Dict[str, Dict[str, List[torch.Tensor]]]:
+    """
+    Same thing as ns.get_matching_activations, but for FX models prepared with
+    this module.
+
+    TODO(future PR): real docblock
+
+    Output format:
+
+    {
+        'layer1.stats': {
+            'name_a': [torch.Tensor(...), ...],
+            'name_b': [torch.Tensor(...), ...],
+        },
+        ...
+    }
+
+    Note, there are three differences from the output format of Eager NS:
+    1. `name_a` and `name_b` are used instead of hardcoding names
+       to `float` and `quantized`.
+    2. Lists of Tensors are returned instead of individual Tensors, to unify
+       the return type for calibrating with 1 input vs N inputs.
+    3. `logger_cls` is included in the API for easy result extraction
+    """
```

Comment:
Just curious why not use get_matching_node_pairs() to find matching activations?

Reply:
yes, that would work too, it would give the same result